extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -161,18 +161,59 @@ private function renderItem( $type, array $data ) {
 			->render( $this->templates[ $type ] );
 	}
 
-	protected function getThumbnailUrl( $url ) {
+	/**
+	 * @desc returns the thumbnail url from
+	 * Vignette or from old service
+	 * @param string $url
+	 * @return string thumbnail url
+	 */
+	protected function getThumbnailUrl( $data ) {
+		$url = $data['url'];
+		// TODO: remove 'if' condition when unified thumb method
+		// will be implemented: https://wikia­inc.atlassian.net/browse/PLATFORM­1237
+		if ( VignetteRequest::isVignetteUrl( $url ) ) {
+			return $this->createVignetteThumbnail( $url );
+		} else {
+			return $this->createOldThumbnail( $data['name'] );
+		}
+	}
+
+	/**
+	 * @param $url
+	 * @return string
+	 */
+	private function createVignetteThumbnail( $url ) {
 		return VignetteRequest::fromUrl( $url )
 			->thumbnailDown()
-			->width( F::app()->checkSkin( 'wikiamobile' ) ?
+			->width(F::app()->checkSkin( 'wikiamobile' ) ?
 				self::MOBILE_THUMBNAIL_WIDTH :
 				self::DESKTOP_THUMBNAIL_WIDTH
 			)
-			->height(self::THUMBNAIL_HEIGHT)
+			->height( self::THUMBNAIL_HEIGHT )
 			->url();
 	}
 
 	/**
+	 * @desc If the image is served from an old
+	 * service we have to again obtain file to
+	 * call the createThumb function
+	 * @param $name
+	 * @return mixed
+	 */
+	private function createOldThumbnail( $name )
+	{
+		$file = wfFindFile( $name );
```

Comment:
consider using one way to get File object across entire extension. Inside `NodeImage.php` you are creating file object using `\WikiaFileHelper::getFileFromTitle($title);`